File: ARRAY/Multiplication/Multiplication.c
#include <stdio.h>

void multiplyMatrices(int m, int n, int p, int A[m][n], int B[n][p], int C[m][p]) {
    for (int i = 0; i < m; i++) {
        for (int j = 0; j < p; j++) {
            C[i][j] = 0;
            for (int k = 0; k < n; k++) {
                C[i][j] += A[i][k] * B[k][j];
            }
        }
    }
}

void displayMatrix(int rows, int cols, int matrix[rows][cols]) {
    for (int i = 0; i < rows; i++) {
        for (int j = 0; j < cols; j++) {
            printf("%d ", matrix[i][j]);
        }
        printf("\n");
    }
}

int main() {
    int m, n, p;
    printf("Enter rows and columns of first matrix: ");
    scanf("%d %d", &m, &n);
    printf("Enter columns of second matrix: ");
    scanf("%d", &p);

    int A[m][n], B[n][p], C[m][p];

    printf("Enter elements of first matrix:\n");
    for (int i = 0; i < m; i++)
        for (int j = 0; j < n; j++)
            scanf("%d", &A[i][j]);

    printf("Enter elements of second matrix:\n");
    for (int i = 0; i < n; i++)
        for (int j = 0; j < p; j++)
            scanf("%d", &B[i][j]);

    multiplyMatrices(m, n, p, A, B, C);

    printf("Resultant Matrix:\n");
    displayMatrix(m, p, C);
    return 0;
}

File: ARRAY/Multiplication/Multiplication.java
import java.util.Scanner;

public class Multiplication {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        // Input matrix dimensions
        System.out.print("Enter rows and columns of first matrix: ");
        int m = sc.nextInt(), n = sc.nextInt();
        System.out.print("Enter columns of second matrix: ");
        int p = sc.nextInt();

        // Validate dimensions
        if (m <= 0 || n <= 0 || p <= 0) {
            System.out.println("Matrix dimensions must be positive integers!");
            return;
        }

        // Initialize matrices
        int[][] A = new int[m][n];
        int[][] B = new int[n][p];
        int[][] C = new int[m][p];

        // Input matrices
        System.out.println("Enter elements of first matrix:");
        inputMatrix(sc, A);
        System.out.println("Enter elements of second matrix:");
        inputMatrix(sc, B);

        // Multiply matrices
        multiplyMatrices(A, B, C);

        // Display result
        System.out.println("Resultant Matrix:");
        displayMatrix(C);

        sc.close();
    }

    // Function to input matrix
    static void inputMatrix(Scanner sc, int[][] matrix) {
        for (int i = 0; i < matrix.length; i++)
            for (int j = 0; j < matrix[0].length; j++)
                matrix[i][j] = sc.nextInt();
    }

    // Function to multiply matrices
    static void multiplyMatrices(int[][] A, int[][] B, int[][] C) {
        int m = A.length, n = A[0].length, p = B[0].length;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < p; j++) {
                C[i][j] = 0;
                for (int k = 0; k < n; k++)
                    C[i][j] += A[i][k] * B[k][j];
            }
        }
    }

    // Function to display matrix
    static void displayMatrix(int[][] matrix) {
        for (int[] row : matrix) {
            for (int val : row)
                System.out.print(val + " ");
            System.out.println();
        }
    }
}


